pkg/apb/secrets.go

```diff
@@ -0,0 +1,159 @@
+package apb
+
+import (
+	"sync"
+
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	logging "github.com/op/go-logging"
+	"github.com/openshift/ansible-service-broker/pkg/clients"
+)
+
+// SecretsConfig - Entry for a secret config block in broker config
+type SecretsConfig struct {
+	Title   string `yaml:"title"`
+	ApbName string `yaml:"apb_name"`
+	Secret  string `yaml:"secret"`
+}
+
+// Validate - Ensures that the secrets config is valid (ie, all strings are non-empty
+func (c SecretsConfig) Validate() bool {
+	for _, str := range []string{c.Title, c.ApbName, c.Secret} {
+		if str == "" {
+			return false
+		}
+	}
+	return true
+}
+
+// AssociationRule - A rule to associate apbs with a secrets
+type AssociationRule struct {
+	apbName string
+	secret  string
+}
+
+type secretsCache struct {
+	mapping map[string][]string
+	rwSync  sync.RWMutex
+	rules   []AssociationRule
+	config  []SecretsConfig
+	log     *logging.Logger
```

Comment:
Yeah, it's very strange and gross. Talked with @shawn-hurley about it and we also concluded that this was just a symptom of the larger logging anti-pattern we have going.